controllers/extendeddaemonset/utils.go

```diff
@@ -60,6 +60,11 @@ func IsCanaryDeploymentEnded(specCanary *datadoghqv1alpha1.ExtendedDaemonSetSpec
 
 // IsCanaryDeploymentPaused checks if the Canary deployment has been paused
 func IsCanaryDeploymentPaused(dsAnnotations map[string]string, ers *datadoghqv1alpha1.ExtendedDaemonSetReplicaSet) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
+	// Unpaused annotation takes precedence
+	isUnpaused, found := dsAnnotations[datadoghqv1alpha1.ExtendedDaemonSetCanaryUnpausedAnnotationKey]
+	if found && isUnpaused == v1alpha1.ValueStringTrue {
+		return false, ""
+	}
```

Comment:
do we still need the `IsCanaryDeploymentUnpaused()` if we consider unpause annotations here? 

or maybe:

```suggestion
	// Unpaused annotation takes precedence
	if IsCanaryDeploymentUnpaused(dsAnnotations) {
		return false, ""
	}
```

Reply:
good question, indeed we may not. will give it a quick test

Reply:
removed! thanks for catching